packages/account/src/providers/provider.ts

```diff
@@ -495,10 +514,87 @@ export default class Provider {
         fullRequest = await options.requestMiddleware(fullRequest);
       }
 
-      return options.fetch ? options.fetch(url, fullRequest, options) : fetch(url, fullRequest);
+      const operationName = fullRequest.body
+        ?.toString()
+        .match(/"operationName":"(.+)"/)?.[1] as keyof SdkOperations;
```

Comment:
Can we improve readability here by making this a helper?

```ts
  private static extractOperationName(body: string | null | undefined): keyof SdkOperations {
    return body?.match(/"operationName":"(.+)"/)?.[1] as keyof SdkOperations;
  }
```

Reply:
You're right that it could be improved. I have refactored the logic dealing with block height into a method in https://github.com/FuelLabs/fuels-ts/pull/3741/commits/4b7c01fc5a05dc4d6c5598df9064d91dd9667685. With this, IMO there's no need to refactor the `operationName` extraction further as the whole code block in the new method is cohesive and straightforward.